bin/adhoc-scripts/fetchFromReqmgr.py

```diff
@@ -0,0 +1,260 @@
+        if not keySetToCheck < wflow.keys():
+            # Create a `tmp` key to fill in all the information we'll need in the follwoing steps.
+            wflow['tmp'] = {}
+
+            # Then append None as the missing values in wflow['tmp']:
+            for key in keySetToCheck:
+                wflow['tmp'][key] = None
+
+            # Then fetch all the data we'd need to calculate the missing values
+            # depending on the type of workflow and add them as key/value pairs in wflow['tmp']:
+            if wflow['RequestType'] == 'ReReco' or \
+               wflow['RequestType'] == 'ReDigi' or \
+               wflow['RequestType'] == 'DQMHarvest':
+                for key in keySetToFetch:
+                    try:
+                        val = wflow[key]
+                    except KeyError:
+                        val = None
+                    wflow['tmp'][key] = val
+            elif wflow['RequestType'] == 'TaskChain':
+                for key in keySetToFetch:
+                    try:
+                        val = wflow['Task1'][key]
+                    except KeyError:
+                        val = None
+                    wflow['tmp'][key] = val
+            elif wflow['RequestType'] == 'StepChain':
+                for key in keySetToFetch:
+                    try:
+                        val = wflow['Step1'][key]
+                    except KeyError:
+                        val = None
+                    wflow['tmp'][key] = val
+            elif wflow['RequestType'] == 'Resubmission':
+                pass
+            else:
+                logger.warning("Found unexpected workflow of type: %s", wflow['RequestType'])
+
+            # Append the High level keys identifying the workflow as key/value pairs in wflow['tmp']:
+            for key in keySetHighLevel:
+                wflow['tmp'][key] = wflow[key]
+
+            # Recalculate the missing values:
+            if wflow['tmp']['SplittingAlgo'] == 'EventBased':
+                logger.info("Found a workflow with EventBased Splitting algorithm:")
+                totalEvents = int(wflow['tmp']['RequestNumEvents'] / wflow['tmp']['FilterEfficiency'])
+                lumisPerJob = math.ceil(wflow['tmp']['EventsPerJob']/wflow['tmp']['EventsPerLumi'])
+                totalLumis = math.ceil(totalEvents / wflow['tmp']['EventsPerLumi'])
+                totalEstimatedJobs = math.ceil(totalLumis/lumisPerJob)
+
+                wflow['tmp']['TotalInputEvents'] = totalEvents
+                wflow['tmp']['TotalInputLumis'] = totalLumis
+                wflow['tmp']['TotalEstimatedJobs'] = totalEstimatedJobs
+                wflow['tmp']['TotalInputFiles'] = 0
+                logger.info('\n%s', pformat(wflow['tmp']))
+
+            elif wflow['tmp']['SplittingAlgo'] == 'EventAwareLumiBased':
+                logger.info("Found a workflow with EventAwareLumiBased Splitting algorithm:")
+                keySetInputLists = {'LumiList', 'RunList', 'BlockList'}
```

Comment:
These parameters do not exist in a ReqMgr2 spec.

Reply:
fixed